### PR TITLE
New module: mmseqs_databases

### PR DESCRIPTION
**Pull request by prototaxites**
Adds MMSEQS_DATABASES, which downloads and formats a protein database for use with mmseqs or metaeuk.

As the source databases are moving targets (and there is no way to specify versions), the test just checks for the existence of the correct output files.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mmseqs/databases/main.nf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    path "mmseqs_database/" , emit: database
+    path "versions.yml"     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    mkdir mmseqs_database/
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/main.nf:23]:
```suggestion
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: 'mmseqs_database`
    """
    mkdir ${prefix}/
```

Just to add some flexibility to the naming of the output :)

File: modules/nf-core/mmseqs/databases/main.nf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    conda "bioconda::mmseqs2=14.7e284"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mmseqs2:14.7e284--pl5321h6a68c12_2':
+        'biocontainers/mmseqs2:14.7e284--pl5321h6a68c12_2' }"
+
+    input:
+    val database
+
+    output:
+    path "mmseqs_database/" , emit: database
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/main.nf:14]:
```suggestion
    path "${prefix}/" , emit: database
```

File: modules/nf-core/mmseqs/databases/main.nf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    mkdir mmseqs_database/
+
+    mmseqs databases \\
+        ${database} \\
+        mmseqs_database/database \\
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/main.nf:27]:
```suggestion
        ${prefix}/database \\
```

File: modules/nf-core/mmseqs/databases/main.nf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    touch mmseqs_database/database.dbtype
+    touch mmseqs_database/database_h
+    touch mmseqs_database/database_h.dbtype
+    touch mmseqs_database/database_h.index
+    touch mmseqs_database/database.index
+    touch mmseqs_database/database.lookup
+    touch mmseqs_database/database_mapping
+    touch mmseqs_database/database.source
+    touch mmseqs_database/database_taxonomy
+    touch mmseqs_database/database.version
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/main.nf:53]:
```suggestion
    prefix = task.ext.prefix ?: 'mmseqs_database'
    """
    mkdir ${prefix}/

    touch ${prefix}/database
    touch ${prefix}/database.dbtype
    touch ${prefix}/database_h
    touch ${prefix}/database_h.dbtype
    touch ${prefix}/database_h.index
    touch ${prefix}/database.index
    touch ${prefix}/database.lookup
    touch ${prefix}/database_mapping
    touch ${prefix}/database.source
    touch ${prefix}/database_taxonomy
    touch ${prefix}/database.version
```

File: modules/nf-core/mmseqs/databases/meta.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/yaml-schema.json
+name: "mmseqs_databases"
+## TODO nf-core: Add a description of the module and list keywords
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/meta.yml:4]: You still have a TODO here ;)

File: modules/nf-core/mmseqs/databases/meta.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  - "mmseqs":
+      description: "MMseqs2: ultra fast and sensitive sequence search and clustering suite"
+      homepage: "https://github.com/soedinglab/MMseqs2"
+      documentation: "https://mmseqs.com/latest/userguide.pdf"
+      tool_dev_url: "https://github.com/soedinglab/MMseqs2"
+      doi: "10.1093/bioinformatics/btw006"
+      licence: "['GPL v3']"
+
+input:
+  #
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/meta.yml:20]:
```suggestion
```

File: modules/nf-core/mmseqs/databases/meta.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+output:
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - database:
+      type: directory
+      description: Directory containing processed mmseqs database
+      pattern: "mmseqs_database/"
```
Comment (nvnieuwk) [inline modules/nf-core/mmseqs/databases/meta.yml:33]:
```suggestion
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Good job! Here are some suggestions to make the module more flexible

File: modules/nf-core/mmseqs/databases/meta.yml (hunk 5 above)
Comment (prototaxites) [reply modules/nf-core/mmseqs/databases/meta.yml]: Whoops 🤦‍♂️ fixed!

**Pull request merged**